osu.Game/Screens/Backgrounds/BackgroundScreenBeatmap.cs

```diff
@@ -35,18 +52,51 @@ public WorkingBeatmap Beatmap
                             Background.FadeOut(250);
                             Background.Expire();
                         }
-
                         b.Depth = newDepth;
-                        AddInternal(Background = b);
+                        FadeContainer.Child = Background = b;
+                        InternalChild = FadeContainer;
                         Background.BlurSigma = BlurTarget;
                     }));
                 });
             }
         }
 
+        public override void OnEntering(IScreen last)
+        {
+            base.OnEntering(last);
+            DimLevel.ValueChanged += _ => updateBackgroundDim();
+            UpdateDim.ValueChanged += _ => updateBackgroundDim();
+            updateBackgroundDim();
+        }
+        public override void OnResuming(IScreen last)
+        {
+            base.OnResuming(last);
+            updateBackgroundDim();
+        }
+
+        public override bool OnExiting(IScreen last)
+        {
+            UpdateDim.Value = false;
+            return base.OnExiting(last);
+        }
+
+        private void updateBackgroundDim()
+        {
+            if (UpdateDim)
+                FadeContainer?.FadeColour(OsuColour.Gray(BackgroundOpacity), 800, Easing.OutQuint);
+            else
+                FadeContainer?.FadeColour(OsuColour.Gray(1.0f), 800, Easing.InQuint);
+        }
+
         public BackgroundScreenBeatmap(WorkingBeatmap beatmap = null)
         {
             Beatmap = beatmap;
+            UpdateDim = new Bindable<bool>();
+        }
+
+        public void EnableUserDim()
```

Comment:
why not just use the bindable directly? as i mentioned in my previous comment: is there any reason to have methods for this?